definitions/npm/ramda_v0.26.x/flow_v0.104.x-/ramda_v0.26.x.js

```diff
@@ -707,13 +707,11 @@ declare module ramda {
   declare function groupBy<T>(fn: (x: T) => string, xs: $ReadOnlyArray<T>): { [key: string]: Array<T>, ... };
   declare function groupBy<T>(fn: (x: T) => string): CurriedFunction1<$ReadOnlyArray<T>, { [string]: Array<T>, ... }>;
 
-  declare function groupWith<T, V: Array<T> | string>(
-    fn: BinaryPredicateFn<T>,
-    xs: V
-  ): Array<V>;
-  declare function groupWith<T, V: Array<T> | string>(
-    fn: BinaryPredicateFn<T>,
-  ): (xs: V) => Array<V>;
+  declare var groupWith: {|
+    (fn: BinaryPredicateFn<string>, string): Array<string>,
+    <A, As: $ReadOnlyArray<A>>(fn: BinaryPredicateFn<A>, As): Array<Array<A>>,
```

Comment:
Reason for not using Array<As> as return type is because we can't wrap a mutable array with read only array, namely, Array<$ReadOnlyArray<A>> is invalid. Correct me if I'm wrong here.

Reply:
Does flow give you an error if you use `Array<$ReadOnlyArray<A>>`?

Reply:
Reading my own comment which I did the other night again, I feel like it's poorly phrased, Flow itself won't complain or break if we make an array of this type:  Array<$ReadOnlyArray<A>>, but in this case, our goal is to allow read only in but mutable out, consider this test case if agreed to be valid one below:
```javascript
  const arr:$ReadOnlyArray<number> = [1, 2, 2, 3, 4, 5, 5]
  const result2:Array<Array<number>> = groupWith((a, b) => a + 1 === b)(arr) 
```
so the final return type, I think, should be  Array<Array<A>>, instead of Array<As> where As is restricted by $ReadOnlyArray<A>, which is also the input type that we want it to be read only.
Using 'As' will give us a 'hybrid' Array type that I think is incorrect as return type.

Reply:
Ahhh, I see. Yeah, I just interpreted what you were saying the wrong way. I think you're right, from the perspective of the output we probably want `Array<Array<A>>`.